x/lockup/keeper/iterator.go

```diff
@@ -47,6 +48,7 @@ func (k Keeper) iteratorLongerDuration(ctx sdk.Context, prefix []byte, duration
 	store := ctx.KVStore(k.storeKey)
 	durationKey := getDurationKey(duration)
 	key := combineKeys(prefix, durationKey)
+	// @TODO: ask for elaboration on how this works
 	// inclusive on longer side, means >= (longer or equal)
```

Comment:
> Thanks for this PR!
> 
> I might have misunderstood issue #756, but seems to me that the direction this PR points to vs the original issue mentioned in the issue seems different.
> 
> How I understood issue #756 was that we can't query total amount of specific denom that's over specific amount of duration, since the query `account-locked-longer-duration-denom` returns all lockups separately instead of adding up the what's in individual lockups.
> 
> This PR, however seems to have the same problem, seems like it's closer to adding a query to query all denoms instead of specific denom over specified time. (I still think this query is superuseful, just wanted to confirm my understanding on the existing issue and the current PR submitted!)

The ticket changed a little due to jitter no longer being an issue and yes I planned todo the same command but for denoms after this one. Rather than do both of them at once I wanted to see the process etc for making a change and what kinds of feedback you would have, and better understand things as a whole, rather than making the same mistakes twice and having to fix them twice in separate PRs. 

tl;dr
Once this goes through I will be making the command you're talking about with `denom` as an additional parameter, this one was easier so decided to give it a go right before its brother which includes `denom`.

Reply:
Do you have discord? I'd like to add you there to discuss further, this is really insightful.

Reply:
Always happy to discuss: `r0man#8579`